### ISSUE TITLE
Whitelist d'ont kick Every Person not whitelisted

### ISSUE THREAD

**Issue by morganm13** [RFX01/PalPal#1]
Whitelist d'ont kick Every Person not whitelisted, sometimes he kick  sometimes not XD

**Comment by morganm13** [comment]
watchdog.rb loop while isn't kick

```
   # Handle Whitelist
if config["PalPalSettings"]["Whitelist"]["Enable"]
    players_joined.each do |player|
        is_whitelisted = false
        config["PalPalSettings"]["Whitelist"]["SteamIDs"].each do |steamid|
            if player["steamid"] == steamid
                is_whitelisted = true
            end
        end
        # Kick player if they're not whitelisted
        unless is_whitelisted
            rcon_exec("KickPlayer #{player["steamid"]}")
            rcon_exec("Broadcast Kicked\xA0\x80non-whitelisted\xA0\x80player\xA0\x80#{player["name"].gsub(" ", "\xA0\x80")}.")
            $logger.warn("Kicked non-whitelisted player: #{player.inspect}")

            # Reevaluate active players after kicking using the new player list
            active_players = new_player_list

            # Log the active players for debugging purposes
            $logger.info("Active players after kick: #{active_players.inspect}")

            # Re-kick any player still not whitelisted
            active_players.each do |active_player|
                unless config["PalPalSettings"]["Whitelist"]["SteamIDs"].include?(active_player["steamid"])
                    rcon_exec("KickPlayer #{active_player["steamid"]}")
                    rcon_exec("Broadcast Re-kicked\xA0\x80non-whitelisted\xA0\x80player\xA0\x80#{active_player["name"].gsub(" ", "\xA0\x80")}.")
                    $logger.warn("Re-kicked non-whitelisted player: #{active_player.inspect}")
                end
            end
        end
    end
end
```

**Comment by RFX01** [comment]
I believe what you're trying to do is kick players retroactively, even if the watchdog wasn't running when the player joined. Because as I stated in the README, players do not get kicked if they are already on the server when PalPal is started.

Your proposed solution would certainly work to an extent, although you're running the code on the join of any random non-whitelisted player, which isn't ideal.

Something else I noticed is that there can be cases where the PalServer RCON returns corrupted SteamIDs, so I'd prefer to have it configurable, since the Steam ID corruption could lead to many unintentional kicks if retroactive kicking is enabled.

Either way, I'll work on implementing Retroactive Kicking later today. I'm also going to add it as a configuration option so people who don't want it can keep it disabled. This should solve your issue without unexpected side effects for anyone else.
